tensorflow_addons/image/dense_image_warp.py

```diff
@@ -48,62 +49,83 @@ def interpolate_bilinear(grid, query_points, indexing="ij", name=None):
-
-        if grid.shape[1] is not None and grid.shape[1] < 2:
-            raise ValueError("Grid height must be at least 2.")
-
-        if grid.shape[2] is not None and grid.shape[2] < 2:
-            raise ValueError("Grid width must be at least 2.")
-
+        # grid shape checks
+        grid_static_shape = grid.shape
         grid_shape = tf.shape(grid)
+        if grid_static_shape.dims is not None:
+            if len(grid_static_shape) != 4:
+                raise ValueError("Grid must be 4D Tensor")
+            if grid_static_shape[1] is not None and grid_static_shape[1] < 2:
+                raise ValueError("Grid height must be at least 2.")
+            if grid_static_shape[2] is not None and grid_static_shape[2] < 2:
+                raise ValueError("Grid width must be at least 2.")
+        else:
+            # pylint: disable=bad-continuation
+            with tf.control_dependencies([
+                    tf.debugging.assert_greater_equal(
+                        grid_shape[1],
+                        2,
+                        message="Grid height must be at least 2."),
+                    tf.debugging.assert_greater_equal(
+                        grid_shape[2],
+                        2,
+                        message="Grid width must be at least 2."),
+                    tf.debugging.assert_less_equal(
+                        tf.cast(
+                            grid_shape[0] * grid_shape[1] * grid_shape[2],
+                            dtype=tf.dtypes.float32),
+                        np.iinfo(np.int32).max / 8.0,
+                        message="The image size or batch size is sufficiently "
+                        "large that the linearized addresses used by "
+                        "tf.gather may exceed the int32 limit.")
+            ]):
+            # pylint: enable=bad-continuation
+                pass
+
+        # query_points shape checks
+        query_static_shape = query_points.shape
         query_shape = tf.shape(query_points)
+        if query_static_shape.dims is not None:
+            if len(query_static_shape) != 3:
+                raise ValueError("Query points must be 3 dimensional.")
+            query_hw = query_static_shape[2]
+            if query_hw is not None and query_hw != 2:
+                raise ValueError("Query points last dimension must be 2.")
+        else:
+            # pylint: disable=bad-continuation
+            with tf.control_dependencies([
+                    tf.debugging.assert_equal(
+                        query_shape[2],
+                        2,
+                        message="Query points last dimension must be 2.")
+            ]):
+            # pylint: enable=bad-continuation
+                    tf.debugging.assert_equal(
+                        query_shape[2],
+                        2,
+                        message="Query points last dimension must be 2.")
+            ]):
```

Comment:
line 103 to line 107 are redundant.

Reply:
thanks. i failed to update things properly using the web tools. should be better now.